src/number_theory/bernoulli_polynomials.lean

```diff
@@ -0,0 +1,199 @@
+/-
+Copyright (c) 2021 Ashvni Narayanan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Ashvni Narayanan
+-/
+
+import number_theory.bernoulli
+
+/-!
+# Bernoulli polynomials
+
+The Bernoulli polynomials are an important tool obtained from Bernoulli numbers.
+
+## Mathematical overview
+
+The $n^ℐrm{th}$ Bernoulli polynomial is defined as
+$$ B_n(X) = ∑_{k = 0}^n {n χoose k} (-1)^k * B_k * X^{n - k} $$
+where $B_k$ is the $k^ℐrm{th}$ Bernoulli number. The Bernoulli polynomials are generating functions,
+$$ t * e^{tX} / (e^t - 1) = ∑_{n = 0}^{∞} B_n(X) * ¼{t^n}{n!} $$
+
+## Implementation detail
+
+The Bernoulli (negative) numbers are first defined, `bernoulli_neg`, in order to make proving
+theorems using Bernoulli polynomials easier.
+
+## Main theorems
+
+- `sum_bernoulli_poly`: The sum of the $k^\mathrm{th}$ Bernoulli polynomial with binomial
+  coefficients up to n is `(n + 1) * X^n`.
+- `exp_bernoulli_poly`: The Bernoulli polynomials act as generating functions for the exponential.
+-/
+
+noncomputable theory
+open_locale big_operators
+open_locale nat
+
```

Comment:
Does it make sense to `open polynomial finset` so that you can abbreviate a lot of the names below (both in statements and proofs)?

Reply:
The problem occurs when both `polynomial` and `power_series` are open, since they have a lot of intersecting lemma names. Since `power_series` is longer than `polynomial`, I thought it might be better to open `power_series` instead. I am happy to change that, if needed.

Reply:
Ok, that makes sense. Just open whichever you use the most.